core/primitives/tipset/file.cpp

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "primitives/tipset/file.hpp"
+
+#include <boost/endian/buffers.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <random>
+
+#include "codec/cbor/light_reader/block.hpp"
+#include "common/file.hpp"
+
+#define BOOL_TRY(...)              \
+  do {                             \
+    if (!(__VA_ARGS__)) return {}; \
+  } while (0)
```

Comment:
Why do  cycle for 1 time need ? Maybe just `if`

Reply:
it's common practice, for `;` and safety
https://gcc.gnu.org/onlinedocs/cpp/Swallowing-the-Semicolon.html